relayer/src/event/rpc.rs

```diff
@@ -178,78 +175,12 @@ pub fn get_all_events(
     Ok(vals)
 }
 
-fn extract_block_events(
-    height: Height,
-    block_events: &HashMap<String, Vec<String>>,
-) -> Vec<(Height, IbcEvent)> {
-    #[inline]
-    fn extract_events<'a, T: TryFrom<RawObject<'a>>>(
-        height: Height,
-        block_events: &'a HashMap<String, Vec<String>>,
-        event_type: &str,
-        event_field: &str,
-    ) -> Vec<T> {
-        block_events
-            .get(&format!("{}.{}", event_type, event_field))
-            .unwrap_or(&vec![])
-            .iter()
-            .enumerate()
-            .filter_map(|(i, _)| {
-                let raw_obj = RawObject::new(height, event_type.to_owned(), i, block_events);
-                T::try_from(raw_obj).ok()
-            })
-            .collect()
-    }
-
-    #[inline]
-    fn append_events<T: Into<IbcEvent>>(
-        events: &mut Vec<(Height, IbcEvent)>,
-        chan_events: Vec<T>,
-        height: Height,
-    ) {
-        events.append(
-            &mut chan_events
-                .into_iter()
-                .map(|ev| (height, ev.into()))
-                .collect(),
-        );
-    }
-
+fn extract_block_events(height: Height, block_events: &[abci::Event]) -> Vec<(Height, IbcEvent)> {
     let mut events: Vec<(Height, IbcEvent)> = vec![];
-    append_events::<ChannelEvents::OpenInit>(
-        &mut events,
-        extract_events(height, block_events, "channel_open_init", "channel_id"),
-        height,
-    );
-    append_events::<ChannelEvents::OpenTry>(
-        &mut events,
-        extract_events(height, block_events, "channel_open_try", "channel_id"),
-        height,
-    );
-    append_events::<ChannelEvents::OpenAck>(
-        &mut events,
-        extract_events(height, block_events, "channel_open_ack", "channel_id"),
-        height,
-    );
-    append_events::<ChannelEvents::OpenConfirm>(
-        &mut events,
-        extract_events(height, block_events, "channel_open_confirm", "channel_id"),
-        height,
-    );
-    append_events::<ChannelEvents::SendPacket>(
-        &mut events,
-        extract_events(height, block_events, "send_packet", "packet_data"),
-        height,
-    );
-    append_events::<ChannelEvents::CloseInit>(
-        &mut events,
-        extract_events(height, block_events, "channel_close_init", "channel_id"),
-        height,
-    );
-    append_events::<ChannelEvents::CloseConfirm>(
-        &mut events,
-        extract_events(height, block_events, "channel_close_confirm", "channel_id"),
-        height,
-    );
+    for abci_ev in block_events {
+        if let Some(ev) = channel_events::try_from_tx(abci_ev) {
```

Comment:
Note that this can in theory extract any channel events, not only the ones that the previous code extracted specifically by matching a key name. So a `WriteAcknowlegement`, `AcknowledgePacket`, or a `TimeoutPacket` could get into the returned list where previous code would ignore these events. I don't know if it is a problem in practice.